tool/cli.js

```diff
@@ -242,16 +242,24 @@ program
     })
   )
 
-  .command("preview", "Open a preview of a slug")
-  .argument("<slug>", "Slug of the document in question")
+  .command("preview", "Open a preview, given a slug or pathname or URL")
+  .argument(
+    "<slugOrPathnameOrUrl>",
+    "Slug or pathname or URL of the document in question"
+  )
   .argument("[locale]", "Locale", {
     default: DEFAULT_LOCALE,
     validator: [...VALID_LOCALES.values()],
   })
   .action(
     tryOrExit(async ({ args }) => {
-      const { slug, locale } = args;
+      const { slugOrPathnameOrUrl, locale } = args;
+      const slug = slugOrPathnameOrUrl.replace(
+        /(https:\/\/([^/]+\/en-US(\/docs)?\/)|files\/en-us\/)?(.+?)(\/index.html)?/g,
+        "$4"
+      );
       const url = `http://localhost:${PORT}${buildURL(locale, slug)}`;
+      console.log(`Previewing files/en-us/${slug}/index.html`.toLowerCase());
```

Comment:
I don't think we need this. It basically says when I typed in already. (if I used a file path)

Reply:
> I don't think we need this. It basically says when I typed in already. (if I used a file path)

OK, I will drop that. (I had added it in consideration for the case where you’ve given it a URL from the production site, and want to view the local equivalent — but I guess in that case you don’t really care what the corresponding path is.)

Reply:
This would anyhow be incorrect when given an url as the slug sanitation is missing.